clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -476,6 +502,10 @@ private IOException translateException(String msg, ApiException e) {
                 return (FileNotFoundException) new FileNotFoundException(msg).initCause(e);
             case HttpStatus.SC_FORBIDDEN:
                 return (AccessDeniedException) new AccessDeniedException(msg).initCause(e);
+            case HttpStatus.SC_INTERNAL_SERVER_ERROR:
+                if (msg.contains("invalid API endpoint")){
```

Comment:
I don't understand, why make this an AccessDeniedException?

Reply:
Mistake.